sympy/polys/polytools.py

```diff
@@ -5307,6 +5307,105 @@ def half_gcdex(f, g, *gens, **args):
         return s, h
 
 
+@public
+def extended_euclidean_algorithm(f, g):
+    """
+    Generator for all intermediate steps in the extended euclidean algorithm applied to polynomials 'f' and 'g'.
+
+    Description
+    ===========
+
+    Implements the extended euclidean algorithm for polynomials as described in e.g.
+        McEliece, R. J., & Shearer, J. B. (1978).
+        A Property of Euclid's Algorithm and an
+        Application to Pade Approximation.
+        SIAM Journal on Applied Mathematics, 34(4), 611-615.
+        doi:10.1137/0134048
```

Comment:
References should go in a section at the end of the docstring:
https://docs.sympy.org/latest/contributing/docstring.html